src/trio/_util.py

```diff
@@ -323,6 +323,20 @@ def __call__(cls, *args: object, **kwargs: object) -> None:
     def _create(cls: type[T], *args: object, **kwargs: object) -> T:
         return super().__call__(*args, **kwargs)  # type: ignore
 
+    def __new__(cls, name: str, bases: tuple[type, ...], namespace: dict[str, Any], **kwargs: object) -> type:  # type: ignore[explicit-any]
+        old_reduce = namespace.get("__reduce__")
+
+        def __reduce__(self: object) -> Any:  # type: ignore[explicit-any]
```

Comment:
In this case, wouldn't `self` be an instance of `NoPublicConstructor`? If so, that would probably alleviate the `attr-defined` error on line 335

Reply:
No, `Cancelled` itself is an instance of `NoPublicConstructor`, not instances of `Cancelled`. Metaclasses are weird!